packages/taiko-client/prover/proof_producer/sgx_producer.go

```diff
@@ -117,13 +129,155 @@ func (s *SGXProofProducer) RequestProof(
 	}, nil
 }
 
+func (s *SGXProofProducer) Aggregate(
+	ctx context.Context,
+	items []*ProofWithHeader,
+	requestAt time.Time,
+) (*BatchProofs, error) {
+	log.Info(
+		"Aggregate sgx batch proofs from raiko-host service",
+		"proofs", items,
+	)
+	if len(items) == 0 {
+		return nil, errors.New("invalid items length")
+	}
+	var (
+		blockIDs []*big.Int
+	)
+	for i, item := range items {
+		blockIDs[i] = item.Meta.GetBlockID()
```

Comment:
Shall we use `append` here for safety?

Reply:
Yes, will scan other code like this error!

Reply:
``` go
blockIDs := make([]*big.Int, len(items))
for i, item := range items {
     blockIDs[i] = item.Meta.GetBlockID()
}
```
I prefer to use this way to reduce memory reallocation and copy creation to become more efficient instead of `append ` when in a high number of `for` loops.

Reply:
`make([]*big.Int, 0, len(items))` is ok, it just allocation once, and after do that `blockIDs`'s length is 0 and capacity is `len(items)`.